### DISCUSSION TITLE
NetBeans 22 won't launch...

### DISCUSSION THREAD

**Discussion by vikingnm** [apache/netbeans#7731] [General]
I installed netbeans 22 and it will not launch. Giving me an error saying I need Java 17 or newer installed. I have Java 22 installed so not understanding why i'm getting the error and why it won't launch. Has anyone run into this and know how to fix it? I'm new to netbeans. This problem isn't showing up in searches.

i created a shortcut and added --jdkhome "C:\Program Files\Java\jdk-22\bin\" but it's saying it can't find Java. Tried all the paths provided from cmd and they all give the not found error

from cmd
java -version
java version "22.0.2" 2024-07-16
Java(TM) SE Runtime Environment (build 22.0.2+9-70)
Java HotSpot(TM) 64-Bit Server VM (build 22.0.2+9-70, mixed mode, sharing)

where java
C:\Program Files\Common Files\Oracle\Java\javapath\java.exe
C:\Program Files (x86)\Common Files\Oracle\Java\java8path\java.exe
C:\Program Files (x86)\Common Files\Oracle\Java\javapath\java.exe
C:\ProgramData\Oracle\Java\javapath\java.exe

echo %JAVA_HOME%
C:\Program Files\Android\Jdk\microsoft_dist_openjdk_1.8.0.25

changed my java home and windows enviroment variable paths to point to the correct folders but still not working

**Comment by Chris2011** [comment]
Maybe you should remove first all java versions that you have, because you have more than one, also JDK 1.8. So let's try this and then try to launch NetBeans again with a clean up to date jdk version.

**Comment by vikingnm** [comment]
Thank you :), The native windows unstiall control panel didn't seem to get all the old versions. I ran their unstall JavaUninstallTool.exe and it seemed to remove it all this time. And reinstalled the jdk and runtimes. Have a few that can't use the 64bit so have to install the 32bit as well but it seems to be working now.

> java -version
java version "22.0.2" 2024-07-16
Java(TM) SE Runtime Environment (build 22.0.2+9-70)
Java HotSpot(TM) 64-Bit Server VM (build 22.0.2+9-70, mixed mode, sharing)

> where java
C:\Program Files\Common Files\Oracle\Java\javapath\java.exe
C:\Program Files (x86)\Common Files\Oracle\Java\java8path\java.exe
C:\Program Files (x86)\Common Files\Oracle\Java\javapath\java.exe
C:\ProgramData\Oracle\Java\javapath\java.exe
C:\Program Files\Java\jdk-22\bin\java.exe

> echo %JAVA_HOME%
C:\Program Files\Java\jdk-22\bin

**Reply by Chris2011** [reply]
Great, happy to help.
